app/main.py

```diff
@@ -1,8 +1,49 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self,
+                 comfort_class: int,
+                 clean_mark: int,
+                 brand: str) -> None:
+        if comfort_class in list(range(1, 8)):
+            self.comfort_class = comfort_class
+        if clean_mark in list(range(1, 11)):
+            self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self,
+                 distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int) -> None:
+        if 1.0 <= distance_from_city_center <= 10.0:
+            self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        if 1.0 <= average_rating <= 5.0:
+            self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list[Car]) -> float:
+        income = 0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income += self.calculate_washing_price(car)
+                self.wash_single_car(car)
+        return round(income, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        cleaning_difficulty = self.clean_power - car.clean_mark
+        single_cleaning_price = (
+            car.comfort_class * cleaning_difficulty
+            * self.average_rating / self.distance_from_city_center
+        )
+        return round(single_cleaning_price, 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        if self.clean_power > car.clean_mark:
```

Comment:
It is not necessary, because we only call this method from serve_cars method and the check is already included

Reply:
Funny the fact that exactly this change was asked in the previous changes request :-)